Server/MirEnvir/Map.cs

```diff
@@ -1103,6 +1103,9 @@ private void CompleteMagic(IList<object> data)
                     value = (int)data[2];
                     location = (Point)data[3];
 
+                    int castId = 0;
+                    if (data.Count >= 5 && data[data.Count - 1] is int ci) castId = ci;
```

Comment:
You don't want to be selecting the last data item, if someone passes through more data items this would break. Check the count but then use the correct index that the castId will always be passed in.